src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
-'use strict';
+('use strict');
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+
+  bite(herbivore) {
+    if (!herbivore.hidden && !(herbivore instanceof Carnivore)) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health <= 0) {
+      const index = Animal.alive.indexOf(herbivore);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
```

Comment:
Instead of using `indexOf` and `splice`, it's recommended to use the `filter` method to remove animals with health less than or equal to 0 from the `Animal.alive` array. This approach avoids nested loops and improves code performance.